### ISSUE TITLE
In the case of SCP commands, SSH.run returns :ok instead of {:ok, output, status}

### ISSUE THREAD

**Issue by brienw** [bitcrowd/sshkit.ex#80]
`SCP.upload` (and download) returns `SSH.run` output which should be `{:ok, output, status}` but it is instead receiving `:ok`

**Comment by klappradla** [comment]
Yes, I also came across this yesterday when figuring out how to test things and was also wondering why it behaves differently.

The way I understand it, this is caused by the special *handler function* `SCP.Uploader.start` passes as an option to `SSH.run` and successively as a function parameter to `Channel.loop/3`. When this special callback [is done](https://github.com/bitcrowd/sshkit.ex/blob/master/lib/sshkit/scp/upload.ex#L59) successfully, it calls [`closed`](https://github.com/bitcrowd/sshkit.ex/blob/master/lib/sshkit/scp/upload.ex#L139) which **only returns a Tuple of two Atoms** - `{:cont, :ok}` in this case.

In contrast to this, the *default*™ handler passed on to `Channel.loop` from `SSH.run` will end the loop with `{:cont, {:ok, Enum.reverse(buffer), status}}` - having a Tuple of an Atom with another Tuple.

Therefore, for the first scenario, the loop ends with `{:done, :ok}` and `SSH.run` will only return `:ok`.
For the second *default*  scenario, the loop ends with `{:done, {:ok, Enum.reverse(buffer), status}}` and `SSH.run` will return the usual `{:ok, output, status}`.

**tl;dr**
I'm pretty new to the codebase and am not 100% sure if I understood everything correctly, but the way it looks to me, I don't see benefit in using a different callback behavior when executing an upload command via `SSH.run`. Instead, it even looks like it would just make the codebase more complex in this are.
Would you prefer to have `SCP.upload`'s behavior aligned with `SSH.run`? If so, I can look into it ✌️

**Comment by pmeinhardt** [comment]
Uh, this is issue is kind of old. But @klappradla got it exactly right:
`SCP.upload`/`SCP.Upload.transfer` use `SSH.run` with a custom handler function.

For uploads and downloads I thought it doesn't make sense to return the same `{:ok, output, exit_code}` tuple since we do not collect any output and exit-code.

I think the documentation is now up-to-date and demos the correct return value. At some point I hope we'll get to adding typespecs as well to further clarify function signatures and allow automated checks using dialyzer.

I'll close this for now, but feel free to open, if you think the return values should be changed 🙂
